src/pointer.rs

```diff
@@ -44,30 +44,30 @@ impl<T: ?Sized> From<Arc<T>> for Pointer<T> {
             let arg = Arc::from_raw(arg);
             let rc = arg.clone();
             mem::forget(arg);
-            NonNull::new_unchecked(Arc::into_raw(rc) as *mut _)
+            arc_to_non_null(rc)
         }
         unsafe fn drop<T: ?Sized>(ptr: NonNull<T>) {
             mem::drop(Arc::from_raw(ptr.as_ptr()));
         }
         Pointer {
-            ptr: unsafe { NonNull::new_unchecked(Arc::into_raw(ptr) as *mut _) },
-            clone: clone::<T>,
-            drop: drop::<T>,
+            ptr: arc_to_non_null(ptr),
+            clone,
+            drop,
             sync: true,
         }
     }
 }
 
 impl<T: ?Sized> From<&'static T> for Pointer<T> {
     fn from(ptr: &'static T) -> Pointer<T> {
-        unsafe fn clone<T: ?Sized>(arg: &T) -> NonNull<T> {
+        fn clone<T: ?Sized>(arg: &T) -> NonNull<T> {
             NonNull::from(arg)
         }
-        unsafe fn drop<T: ?Sized>(_ptr: NonNull<T>) { }
+        fn drop<T: ?Sized>(_ptr: NonNull<T>) { }
         Pointer {
             ptr: NonNull::from(ptr),
-            clone: clone::<T>,
-            drop: drop::<T>,
+            clone,
```

Comment:
Tried hard to define this in terms of `NonNull::from` but I couldn't get the type to aligned.  Read a bunch about higher-ranked trait bounds but couldn't figure it out.